test/extended/oauth/client_secret.go

```diff
@@ -136,17 +148,24 @@ func bodyClientAuthRequest(tokenEndpoint string) *http.Request {
 	)
 	o.Expect(err).NotTo(o.HaveOccurred())
 
-	return req
+	return []*http.Request{req}
 }
 
-func headerClientAuthRequest(tokenEndpoint string) *http.Request {
-	req, err := http.NewRequest("POST", tokenEndpoint,
+func headerClientAuthRequest(tokenEndpoint string) []*http.Request {
+	reqNoURLEscape, err := http.NewRequest("POST", tokenEndpoint,
 		bytes.NewBufferString("grant_type=authorization_code&"+
 			"code="+authzToken,
 		),
 	)
 	o.Expect(err).NotTo(o.HaveOccurred())
+	reqNoURLEscape.Header.Set("Authorization", fmt.Sprintf("Basic %s", base64.StdEncoding.EncodeToString([]byte("oauth-client-with-plus:secret+with+plus"))))
```

Comment:
Please use https://pkg.go.dev/net/http#Request.SetBasicAuth for setting basic auth